core/src/main/java/org/yggard/brokkgui/skin/GuiTextfieldSkin.java

```diff
@@ -1,23 +1,102 @@
 import org.yggard.brokkgui.element.GuiTextfield;
+import org.yggard.brokkgui.internal.IGuiHelper;
 import org.yggard.brokkgui.internal.IGuiRenderer;
 import org.yggard.brokkgui.paint.Color;
 import org.yggard.brokkgui.paint.RenderPass;
 import org.yggard.brokkgui.validation.BaseTextValidator;
 
+import fr.ourten.teabeans.binding.BaseBinding;
+import fr.ourten.teabeans.value.BaseProperty;
+
 /**
  * @author Ourten 2 oct. 2016
  */
 public class GuiTextfieldSkin<T extends GuiTextfield> extends GuiBehaviorSkinBase<T, GuiTextfieldBehavior<T>>
 {
+    
+    private final BaseProperty<String> ellipsedPromptProperty;
+    private final BaseProperty<Integer> displayOffsetProperty;
+    // Here to keep last value of the displayOffsetProperty
+    private int displayOffset = 0;
+    private final BaseProperty<String> displayedTextProperty;
+    
     public GuiTextfieldSkin(final T model, final GuiTextfieldBehavior<T> behaviour)
     {
         super(model, behaviour);
 
         this.getModel().getStyle().registerProperty("-text-color", Color.WHITE, Color.class);
+        
+        this.ellipsedPromptProperty = new BaseProperty<>("", "ellipsedPromptProperty");
+        this.displayOffsetProperty = new BaseProperty<>(0, "displayOffsetProperty");
+        this.displayedTextProperty = new BaseProperty<>("", "displayedTextProperty");
+        
+        this.ellipsedPromptProperty.bind(new BaseBinding<String>()
+        {
+            {
+                super.bind(getModel().getPrompTextProperty(), getModel().getPromptEllipsisProperty(),
+                        getModel().getTextPaddingProperty(), getModel().getWidthProperty());
+            }
+            
+            @Override
+            public String computeValue() {
+                return trimTextToWidth(getModel().getPromptText(), getModel().getPromptEllipsisProperty().getValue(),
+                        (int) (getModel().getWidth() - getModel().getTextPaddingProperty().getValue() * 2), 
+                        BrokkGuiPlatform.getInstance().getGuiHelper());
+            }
+            
+        });
+        
+        displayOffsetProperty.bind(new BaseBinding<Integer>() 
+        {
+            
+            {
+                super.bind(getModel().getCursorPositionProperty(), getModel().getTextPaddingProperty(), getModel().getWidthProperty());
+            }
+
+            @Override
+            public Integer computeValue() 
+            {
+                if(getModel().getCursorPosition() <= displayOffset)
+                {
+                    displayOffset = getModel().getCursorPosition();
+                    return displayOffset;
+                }
+                IGuiHelper helper = BrokkGuiPlatform.getInstance().getGuiHelper();
+                while(helper.getStringWidth(getModel().getText().substring(
+                        displayOffset, 
+                        getModel().getCursorPosition())) > getModel().getWidth() - 2 * getModel().getTextPaddingProperty().getValue())
```

Comment:
`getModel().getTextPaddingProperty().getValue()` should be replaced with `getModel().getTextPadding()` to improve readability.